src/filesystem.cpp

```diff
@@ -20,6 +20,34 @@
 #endif
 
 static const std::string invalid_chars = "\\/:?\"<>|";
+static const std::array invalid_names = {
+    std::string_view( "CON" ),
+    std::string_view( "PRN" ),
+    std::string_view( "AUX" ),
+    std::string_view( "NUL" ),
+    std::string_view( "COM0" ),
+    std::string_view( "COM1" ),
+    std::string_view( "COM2" ),
+    std::string_view( "COM3" ),
+    std::string_view( "COM4" ),
+    std::string_view( "COM5" ),
+    std::string_view( "COM6" ),
+    std::string_view( "COM7" ),
+    std::string_view( "COM8" ),
+    std::string_view( "COM9" ),
+    std::string_view( "COM" ),
```

Comment:
```suggestion
    std::string_view( "COM¹" ),
    std::string_view( "COM²" ),
    std::string_view( "COM³" ),
```
I tried and `COM` is fine while `COM¹` etc are actually not valid. (Why Windows reserves names with unicode characters for some ancient system is another question but anyway...)

Reply:
@Qrox: COM is well documented as not being valid. Even if your specific Windows version and locale allows it, it should be not allowed

Reply:
@Zireael07 You're probably thinking of `CON`? `COM` is not listed in the documentation I referred to, only `COM` followed by a digit. Do you have a source that says `COM` is not valid under specific Windows versions/locales?